From e8a5464b0f767cfd2d0f5b8c8dae0d874d0c591d Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sat, 17 Jan 2015 19:08:57 -0800
Subject: [PATCH 1004/1006] Only call setpriority(PRIO_DARWIN_THREAD, 0,
 PRIO_DARWIN_BG) if it is available

Tiger and earlier versions of darwin do not support this.

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit a186e40307ffb706d39353bfc285c12d242e26ee)
---
 tools/libclang/CIndex.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git llvm_release_39/tools/clang/tools/libclang/CIndex.cpp macports_release_39/tools/clang/tools/libclang/CIndex.cpp
index 027bf95..8081753 100644
--- llvm_release_39/tools/clang/tools/libclang/CIndex.cpp
+++ macports_release_39/tools/clang/tools/libclang/CIndex.cpp
@@ -7806,7 +7806,7 @@ void clang::setThreadBackgroundPriority() {
   if (getenv("LIBCLANG_BGPRIO_DISABLE"))
     return;
 
-#ifdef USE_DARWIN_THREADS
+#if defined(USE_DARWIN_THREADS) && defined(PRIO_DARWIN_THREAD) && defined(PRIO_DARWIN_BG)
   setpriority(PRIO_DARWIN_THREAD, 0, PRIO_DARWIN_BG);
 #endif
 }
-- 
2.10.1

